crates/oxc_minifier/src/peephole/remove_unused_declaration.rs

```diff
@@ -11,6 +11,25 @@ impl<'a> PeepholeOptimizations {
             && !ctx.scoping().root_scope_flags().contains_direct_eval()
     }
 
+    fn can_explode_expression_as_array_pattern(expr: &Expression<'a>, ctx: &Ctx<'a, '_>) -> bool {
+        match expr {
+            Expression::ArrayExpression(_)
+            | Expression::StringLiteral(_)
+            | Expression::TemplateLiteral(_) => true,
+            Expression::Identifier(ident) => {
+                ident.name == "arguments"
+                    && ctx.is_global_reference(ident)
+                    && ctx.current_scope_flags().is_function()
```

Comment:
I guess we need to crawl up until we find a function scope or the top level scope to handle cases like
```js
function foo() {
  console.log((() => arguments)())
}
foo()
```